### DISCUSSION TITLE
htmlarea

### DISCUSSION THREAD

**Discussion by bbannon** [craigk5n/webcalendar#276] [General]
Is anyone still using htmlarea?

**Comment by craigk5n** [comment]
Yes, we still do.  It is no longer something a user needs to install.  Instead we use the CDN version if the "HTML in Description" setting is set to true.  See around line 340 in includes/init.php.  I'd like to get away from using CDN references in the code since it stops working if you lose your external internet connection.

**Reply by bbannon** [reply]
OK. Then I suggest, use the CDN the first time and save the file on the user's system. Preferably in a  subfolder of "vendor" or "3rdparty" with phpmailer, jquery and other things we use but don't maintain.
 Set up a variable in default_config with the path to the user's copy, if available, or the CDN.

**Comment by s22-tech** [comment]
Why not just include it with the zip file?  That way the version can be controlled, while preventing any 3rd party script from "phoning home".

**Comment by craigk5n** [comment]
To be consistent with the way we are now handling bootstrap and jquery, we should:
- add ckeditor to composer.json and let composer download CKEditor for us
- update `Makefile` to copy over only the stuff we need into the `pub` directory (we only commit stuff in `pub` to our git repo and ignore everything that composer downloads into `vendor`)
- update `includes/init.php` to include CKEditor when needed.
But which version of CKEditor to include (4 or 5)?  (Which can we get via composer?)
And which bundle (minimal, standard or full)?

**Reply by bbannon** [reply]
I don't know anything about composer, but it sounds like a good idea.
Last time I looked at ckeditor, they were trying to go with a free module and a paid module. I'm guessing we want the free one. And minimal should be plenty.
